### PR TITLE
foldit: cask updates

### PR DESCRIPTION
**Pull request by jhed9**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---
The existing cask installs an old version of Foldit. The URL needs to be updated.

I also fixed the livecheck, added a little to the description, and added the auto_updates tag.

### CODE REVIEW

File: Casks/f/foldit.rb [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-    url :url
-    strategy :extract_plist
+    url "https://fold.it/releases"
+    regex(/<h2[^>]*?>[^<]*?(\d+)/i)
+    strategy :page_match do |page, regex|
+      page.scan(regex).map { |match| (match[0]).to_s }
+    end
   end
 
+  auto_updates true
```
Comment (bevanjkay) [inline Casks/f/foldit.rb:19]: Does this actually update to the latest version, or just update some of the contents of the application? it seems to update some internal files, but not the whole application.

Comment (jhed9) [reply Casks/f/foldit.rb:19]: You are correct, it doesn't replace the whole app bundle, it only modifies/adds files inside of the app bundle when an update comes out. It is kind of unusual in that way. Nonetheless, after it updates the files inside the bundle, the app reports the new version as installed. It also does the updates automatically when they are released.

File: Casks/f/foldit.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
...
   homepage "https://fold.it/"
 
   livecheck do
-    url :url
-    strategy :extract_plist
+    url "https://fold.it/releases"
+    regex(/<h2[^>]*?>[^<]*?(\d+)/i)
+    strategy :page_match do |page, regex|
+      page.scan(regex).map { |match| (match[0]).to_s }
+    end
```
Comment (bevanjkay) [inline Casks/f/foldit.rb:16]:
```suggestion
```

The strategy block can be removed, as this is the default behaviour.

**Review by bevanjkay — CHANGES_REQUESTED** [review]

File: Casks/f/foldit.rb [hunk 3]
```diff
@@ -1,17 +1,22 @@
...
-  version "1.0"
+  version "33"
   sha256 :no_check
 
-  url "https://fold.it/portal/files/app/Foldit-macos_x64.dmg"
+  url "https://files.ipd.uw.edu/pub/foldit/Foldit-macos_x64.dmg",
+      verified: "files.ipd.uw.edu/pub/foldit/"
   name "Foldit"
-  desc "Crowdsourcing computer game"
+  desc "Crowdsourcing, protein folding computer game"
```
Comment (razvanazamfirei) [inline Casks/f/foldit.rb:8]:
```suggestion
  desc "Protein folding computer game"
```

**Review by razvanazamfirei — APPROVED** [review]

**Review by bevanjkay — APPROVED** [review]

**Comment by bevanjkay** [comment]
Thank you @jhed9

**Pull request merged**